[Osquery] Add docs for Osquery API (#137162)
elastic/kibana: docs/api/osquery-manager/saved-queries/update.asciidoc

[[osquery-manager-saved-queries-api-update]]
=== Update saved query API
++++
<titleabbrev>Update saved query</titleabbrev>
++++

experimental[] Update saved queries.

> WARNING: You cannot update a prebuilt saved query (`prebuilt = true`).

Reviewer: ```suggestion
WARNING: You are unable to update a prebuilt saved query (`prebuilt = true`).
```

[[osquery-manager-saved-queries-api-update-request]]
==== Request

`PUT <kibana host>:<port>/api/osquery/saved_queries/<id>`

`PUT <kibana host>:<port>/s/<space_id>/api/osquery/saved_queries/<id>`

[[osquery-manager-saved-queries-api-update-path-params]]
==== Path parameters

`space_id`::

>   (Optional, string) An identifier for the space. If `space_id` is not provided in the URL, the default space is used.

Reviewer: ```suggestion
  (Optional, string) The space identifier. When `space_id` is not provided in the URL, the default space is used.
```

`id`::
  (Required, string) The ID of the saved query you want to update.

[[osquery-manager-saved-queries-api-update-body-params]]
==== Request body

> `id`:: (Required, string) The name of the saved query.

Reviewer: ```suggestion
`id`:: (Required, string) The saved query name.
```

> `description`:: (Optional, string) A description of the saved query.

Reviewer: ```suggestion
`description`:: (Optional, string) The saved query description.
```

> `platform`:: (Optional, string) Restrict this query to a given platform, default is 'all' platforms; you may use commas to set multiple platforms (e.g. 'linux,darwin').

Reviewer: ```suggestion
`platform`:: (Optional, string) Restricts the query to a specified platform. The default is 'all' platforms. To specify multiple platforms, use commas. For example, 'linux,darwin'.
```

> `query`:: (Required, string) The SQL query to run.

Reviewer: ```suggestion
`query`:: (Required, string) The SQL query you want to run.
```

> `version`:: (Optional, string) Only run on osquery versions greater than or equal-to this version string

Reviewer: ```suggestion
`version`:: (Optional, string) Runs on Osquery versions greater than or equal to the specified version string.
```

> `internal`:: (Optional, string) An interval in seconds to run the query.

Reviewer: ```suggestion
`internal`:: (Optional, string) The interval, in seconds, to run the query.
```

> `ecs_mapping`:: (Optional, object) Config allows to map osquery results columns or static values to ECS fields.

Reviewer: ```suggestion
`ecs_mapping`:: (Optional, object) Maps Osquery result columns or static values to ECS fields.
```
